csharp/src/SeedLang/Visualization/EventData.cs

```diff
@@ -0,0 +1,101 @@
+// Copyright 2021-2022 The SeedV Lab.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Text;
+
+namespace SeedLang.Visualization {
+  public enum VariableType {
+    Global,
+    Local,
+    Upvalue,
+  }
+
+  // The class that is used as operands of binary and comparison events. It could be a variable,
+  // an element of containers or a temporary value. The Variable field is null if it's a temporary
+  // value.
+  public class Operand {
+    public bool IsVariable => !(Variable is null);
```

Comment:
The structure of `Operand` and `Variable` is a bit confusing:

First, the current `Operand` is classified to:
- Is a Variable
- Is NOT a Variable (that is, a temporary value)

Second, the current `Variable` is classified to:
- Is an Element of a Container
- Is NOT an Element of a Container (that is, a normal variable)

Theoretically "an element of a container" is not always treated as a variable, right?

Can we define an Enum with three categories to classify an operand?
- Variable operand
- ElementOfContainer operand
- TemporaryValue operand

Reply:
Fixed. Please see my comments seperately.